Allow build with nix-tools.

jbgi opened the pull request:
Needs to be done in this PR:
- [x] Hydra jobset for Linux.
- [x] Hydra jobset for MacOS.
- [x] Builds of cardano-http-bridge, jörmungandr, cardano-sl-node on linux.
- [x] Nix-shell development environment with all dependencies installed.
- [x] Buildkite job to check hydra evaluation. @cleverca22 
- [x] Buildkite job to regenerate nix and provide diff. input-output-hk/iohk-nix#88

Can do it later:
- [ ] Cross builds for windows - some dependencies are not building yet. @angerman 
- [ ] Benchmarks building @angerman @rvl
- [ ] Builds of cardano-http-bridge, jörmungandr on MacOS. @cleverca22 @disassembler 
- [ ] Integration tests passing.

[Hydra jobset](https://hydra.iohk.io/jobset/Cardano/cardano-wallet-pr-103)

jbgi committed:
Use nix-tools.

rvl commented on stack.yaml:
```diff
@@ -1,4 +1,5 @@
 resolver: lts-13.8
+compiler: ghc-8.6.3
```
Why is this needed, if LTS 13.8 already implies GHC 8.6.3?

rvl commented on nix/stack-shell.nix:
```diff
@@ -3,7 +3,7 @@
 , config ? {}
 , walletPackages ? import ./.. { inherit config system; }
-, pkgs ? walletPackages.pkgs
+, pkgs ? walletPackages._lib.pkgs
```
Can `iohk-nix` be improved here? Getting the pinned pkgs seems like a fairly standard thing to do, so there should be a "standard" way to do it.

rvl commented on release.nix:
```diff
@@ -0,0 +1,52 @@
+  required-targets = jobs: [
+    # targets are specified using above nomenclature:
+    jobs.nix-tools.tests.cardano-wallet.unit.x86_64-linux
+    jobs.nix-tools.tests.cardano-wallet.integration.x86_64-linux
```
As they are, the integration tests probably won't work in a sandboxed build.

rvl commented on release.nix:
```diff
@@ -0,0 +1,52 @@
+    jobs.nix-tools.tests.cardano-wallet.unit.x86_64-linux
+    jobs.nix-tools.tests.cardano-wallet.integration.x86_64-linux
+
+    jobs.nix-tools.exes.cardano-wallet.x86_64-linux
```
The wallet needs `cardano-http-bridge` in its `PATH` - can something be done about it?

rvl commented on default.nix:
```diff
@@ -1,20 +1,50 @@
+{ ... }@args:
+# We will instantiate the default-nix template with the
+# nix/pkgs.nix file...
+commonLib.nix-tools.default-nix ./nix/pkgs.nix args
```
This `commonLib.nix-tools.default-nix` function is very obscure. I would prefer it if `default.nix` did not use that function, even if it means that some code has been repeated.

rvl commented on nix/stack-shell.nix:
```diff
@@ -3,7 +3,7 @@
 { system ? builtins.currentSystem
```
When I try `stack --nix build`, it starts to build GHC 8.4.3 (`/nix/store/7sih6g9zyzbrjc0qqgq5692m3pwgazkz-ghc-8.4.3.drv`). Why is that?

rvl reviewed (dismissed):
Thanks for working on this @jbgi.

Does the windows build work?

rvl committed:
Update the haskell.nix code

rvl committed:
Add lorri/direnv file

rvl committed:
Cabal files: Single LICENSE text file, refer to license by name

The Nix haskell builder is having problems with symlinks to
directories above the package root, because these are not included in
the package sdist.

rvl committed:
Cabal files: Add hspec-discover build-tool dependency

rvl committed:
Add cabal.project

cleverca22 committed:
add check-hydra.sh

cleverca22 commented on default.nix:
```diff
@@ -1,20 +1,56 @@
 
 # Keep this argument even if unused.
 # It will prevent Hydra from caching the evaluation.
 , gitrev ? iohkLib.commitIdFromGitRepo ./.
```
this will only trigger a new eval if its actually an input to a job

it needs to be routed to the forceNewEval job that iohk-nix is generating

rvl replied:
Good point. I have removed it from here, and made it local to the `required` aggregate job.

rvl committed:
Buildkite: Add check-nix-tools
